A14-channelz.md

```diff
@@ -0,0 +1,1051 @@
+there are multiple clients and servers.  To avoid confusion, a strict
+terminology will be used:
+
+1.  A "channel" represents an abstraction that can start and complete RPCs.  A
+    channel may have channels and subchannels.  A channel is a directed 
+    acyclic graph (DAG).  Only leaf channels may have
+    sockets.  Interior channels may only have channels and subchannels.
+2.  A "subchannel" represents an abstraction that is load balanced over by an 
+    owning channel. A subchannel may have channels and subchannels.  A 
+    subchannel is a directed acyclic graph (DAG).  Only leaf subchannels may 
+    have sockets.  Interior subchannels may only have channels and subchannels.
+3.  A descendent channel is either a channel or subchannel that is logically 
+    owned by a higher level channel.  Typically these are subchannels, but may
+    be channels themselves. 
+4.  A "server" represents the entry point for RPCs.  A server may have one or
+    more listening sockets, and has a collection of "services".  Unlike 
+    clients, servers are not hierarchical.  A server may only have sockets.
+5.  An "endpoint" is either a channel or a server.  The local and remote
+    endpoints may exist within the same process.
+6.  A "socket" is roughly equivalent to a file descriptor.  A socket may be
+    connected or unconnected (such as a listen socket).
+7.  A "connection", sometimes called a transport, represents the link between
+    two endpoints.  It is included here for completeness.
+8.  A "client" is a role rather than an abstraction.  A client is the initiator
+    of connections and RPCs, via its channels.  A client has channels, which it
+    uses to send RPCs. This term is used less commonly.
+
+
+# Channelz Data
+
+The channelz service exposes internal implementation details in a public way.
+Since different implementations have different internal details, the channelz
+data is lightly abstracted.  There are four top level entities exposed:
+Channels, Subchannels, Servers, and Sockets.
+
+Each entity is uniquely identified by a positive integer, called the id.  This
+id must be distinct and may not be reused over the lifetime of the process for
+any entity.  This ensures that querying channelz will never result in referring
+to the wrong entity.  While ids are not allocated sequentially, they are
+ordered.  This enables pagination of the results when querying channelz.
+Since protobuf treats the number 0 as the default value, it is unused as an id.
+
+Along with the id, a human readable name can optionally be associated with each
+the id.  The id and the name together form a "reference".  Thus, Channels,
+Servers and Sockets are identified by their respective references.  These
+references are abbreviated ChannelRefs, SubchannelRefs, ServerRefs, and 
+SocketRefs.  Note that only the id is necessary to query channelz.
+
+The data representation of each ref:
+
+![message ref structure][message ref structure]
```

Comment:
The content of this diagram is just text.  Given that, why put it in a diagram instead of inlining it directly into this document?